Node continues to publish last frame after end of video

I'm using this package to stream a video as a stream of frames to a node which manipulates the frames and saves them to disk. I launch the `camera.launch` from this package in another launch file like this:

```
<include file="$(find video_stream_opencv)/launch/camera.launch">
	<arg name="camera_name" value="$(arg video_stream_topic_name)"/>
	<arg name="video_stream_provider" value="$(arg video_file)"/>
	<arg name="stop_frame" value="-1"/>
</include>
```

Once the video is finished (~1000 frames), it continues to publish the last frame indefinitely, while throwing the following error on console:

```
[ERROR] [1590444387.369692670]: Could not capture frame
[ERROR] [1590444387.402886483]: Could not capture frame
[ERROR] [1590444387.436512947]: Could not capture frame
[ERROR] [1590444387.470821575]: Could not capture frame
[ERROR] [1590444387.504174047]: Could not capture frame
```

--- a/src/video_stream.cpp
+++ b/src/video_stream.cpp
@@ -187,18 +187,16 @@ virtual void do_publish(const ros::TimerEvent& event) {
     }
 
     // Check if grabbed frame is actually filled with some content
-    if(!frame.empty()) {
+    if(!frame.empty() && is_new_image) {
         // From http://docs.opencv.org/modules/core/doc/operations_on_arrays.html#void flip(InputArray src, OutputArray dst, int flipCode)
         // FLIP_HORIZONTAL == 1, FLIP_VERTICAL == 0 or FLIP_BOTH == -1
         // Flip the image if necessary
-        if (is_new_image){
-          if (latest_config.flip_horizontal && latest_config.flip_vertical)
-            cv::flip(frame, frame, -1);
-          else if (latest_config.flip_horizontal)
-            cv::flip(frame, frame, 1);
-          else if (latest_config.flip_vertical)
-            cv::flip(frame, frame, 0);
-        }
+        if (latest_config.flip_horizontal && latest_config.flip_vertical)
+          cv::flip(frame, frame, -1);
+        else if (latest_config.flip_horizontal)
+          cv::flip(frame, frame, 1);
+        else if (latest_config.flip_vertical)
+          cv::flip(frame, frame, 0);
         cv_bridge::CvImagePtr cv_image =
           boost::make_shared<cv_bridge::CvImage>(header, "bgr8", frame);
         if (latest_config.output_encoding != "bgr8")

